### DISCUSSION TITLE
AOC Day 15: The Bronx is Up and the Battery is Down

### DISCUSSION THREAD

**Discussion by rhdeck** [Virtual-Coffee/virtualcoffee.io#777] [Advent of Code 2022]
Discussion of solutions for [Advent of Code Day 15](https://adventofcode.com/2022/day/15) here.

Don't forget about the VC Advent of Code private leaderboard:

Go to [leaderboard/private](https://adventofcode.com/2022/leaderboard/private)
Use join code 363496-e6f42283

**Comment by rhdeck** [comment]
My naive part 2 that extended part 1 across multiple lines was projected to take 10 days! Proud of my [solution](https://github.com/rhdeck/aoc-2022/blob/main/src/15/day_15.ipynb), which takes ~~40~~30s

**Reply by tkshill** [reply]
I think we have basically the same approaches, using our respective language idiomatic mechanics. Similar solve times as well. And it feels like you're becoming the numpy expert at this point! Nice solve!

**Comment by doleraj** [comment]
This code, it is not the prettiest. But it works, and it's zippy, and I made it on my own. https://github.com/doleraj/adventofcode/blob/main/aoc2022/src/day15/index.ts

**Reply by tkshill** [reply]
Lovely! Loving the use of classes and interfaces here to keep context. With all of the moving parts of this problem, TS seems like a much better experience than JS. I want compiler assistance.

**Comment by tkshill** [comment]
I am suuuuper proud of this one. Calmly thought through my solves, solutions _mostly_ worked on the first try as well. Anticipated the explosion on part2 so didn't write a line of code until I felt like I had something efficient.

Actual amount of code is very minimal. I am pleased. 

[solve](https://github.com/tkshill/AdventOfCode2022/blob/main/AOC2022/Day15.fs)
